R/timer.R

```diff
@@ -96,3 +132,29 @@ scheduleTask <- function(millis, callback) {
     invisible(timerCallbacks$unschedule(id))
   }
 }
+
+#' Get a scheduler function for scheduling tasks. Give priority to the
+#' session scheduler, but if it doesn't exist, use the global one.
+#' @noRd
+defineScheduler <- function(session){
+  if (!is.null(session)){
+    if (!is.null(session$scheduleTask)){
+      return(session$scheduleTask)
+    }
+  }
+  scheduleTask
+}
+
+
+#' Get the current time a la `Sys.time()`. Prefer to get it via the
+#' `session$now()` function, but if that's not available, just return the
+#' current system time.
+#' @noRd
+getTime <- function(session){
+  if (!is.null(session)){
```

Comment:
Minor nit: I think it's slightly clearer to have a single `if` statement with `&&`, since there's no complicated logic inside.